frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -406,16 +414,41 @@ class SegmentsView extends React.Component<Props, State> {
         searchableTreeItemList.push(item);
       });
 
-      return {
+      updateStateObject = {
         groupTree: generatedGroupTree,
         searchableTreeItemList,
         prevProps: nextProps,
       };
-    } else {
+    }
+    if (prevState.prevProps?.isosurfaces !== isosurfaces) {
+      //if any segment is invisible, set visibility false, so that the first action is to make all meshes visible
```

Comment:
```suggestion
      // Derive the areSegmentsInGroupVisible state so that we know per group
      // if it contains only visible elements. This is used to know whether "Show meshes" or
      // "Hide meshes" context item should be shown.
      // If any segment is invisible, set the visibility of the group to false, so that the preferred
      // action is to make all meshes visible.
```